src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx

```diff
@@ -180,7 +180,10 @@ export default class OrganizationDiscover extends React.Component {
     this.runQuery();
   };
 
-  runQuery = () => {
+  runQuery = e => {
+    // This can be called from an event handler (e.g. on form submit)
+    e?.preventDefault();
```

Comment:
If we want to `preventDefault` alongside `runQuery` in just one place, I think it's cleaner to just write a function that does that then calls this rather than change the signature here and always need to check for `e`